docs/utils/useJsonColor.ts

```diff
@@ -70,22 +73,53 @@ export const useJsonColor = (): CompiledColors => {
       Object.entries(themes).forEach(([themeName, theme]) => {
         const tempThemeColors: Theme = {};
         Object.entries(theme).forEach(([colorName, colorValues]) => {
-          const tempThemeColorPalette: ColorPalette = {};
-          Object.entries(colorValues).forEach(([shade, details]) => {
-            const { value, description } = details;
-            const resolvedValue = parseColorValue(value, figmaBrandColors);
-            const tempShadeColor = {
-              ...details,
-              value: resolvedValue,
-              description:
-                description + (value === resolvedValue ? '' : ` ${value}`),
+          if (typeof colorValues.value === 'string') {
+            tempThemeColors[colorName] = {
+              ...colorValues,
+              value: parseColorValue(colorValues.value, figmaBrandColors),
             };
-            tempThemeColorPalette[shade] = tempShadeColor;
-          });
-          tempThemeColors[colorName] = tempThemeColorPalette;
+          } else {
+            const tempThemeColorPalette: ColorPalette = {};
+            Object.entries(colorValues).forEach(([shade, details]) => {
+              let resolvedValue = parseColorValue(
+                details.value,
+                figmaBrandColors,
+              );
+              if (!isHexColor(resolvedValue)) {
+                const cleanResolvedValue = resolvedValue
+                  .slice(1, -1)
+                  .split('.'); // Split the reference into parts
+                const category = cleanResolvedValue[0]; // Get the category (e.g., 'text')
+                const key = cleanResolvedValue[1]; // Get the key (e.g., 'default')
+                if (category) {
+                  if (theme[category]?.[key]) {
+                    resolvedValue = parseColorValue(
+                      theme[category][key].value,
```

Comment:
There are a couple of TypeScript errors here that will need to be resolved in future
`Type 'undefined' cannot be used as an index type`

![Screenshot 2024-05-27 at 3 13 46 PM](https://github.com/MetaMask/design-tokens/assets/8112138/11e0c245-3e8b-47db-8475-02b546a553e1)